### ISSUE TITLE
[SuperEditor] FEAT: Tables

### ISSUE THREAD

**Issue by raulmateolabhouse** [superlistapp/super_editor#953]
My company and I are looking for a Rich editor that supports tables. As far as I am concerned, FlutterQuill and SuperEditor both lack this feature; so I wanted to open a conversation about the feasibility to implement this feature on SuperEditor.

# Description of the feature

We do not want a GridEditor. We are looking for a Single Column Editor that allows to insert tables in an specific block, just as Markdown allows.

| This | is   | a     | table |
|------|------|-------|-------|
| 3232 | 3232 | 32323 | 3232  |

We would also like to style the border colors.

# Questions

- Does Super Editor support this feature? (I am assuming no, but just in case)
- How hard would it be, based on the state of the project, to implement this feature?
- Are other companies interested in this feature as well? Is there someone investing on it already?
- Would be feasible for an external team to add this feature on the project, or may be fork it?

**Comment by matthew-carroll** [comment]
Hi @raulmateolabhouse. Thanks for asking about tables. Here are answers to your questions:

> Does Super Editor support this feature? (I am assuming no, but just in case)

Super Editor doesn't have a table component. However, we expect that Super Editor's foundation should support such a feature. We've discussed it in the past.

> How hard would it be, based on the state of the project, to implement this feature?

To answer this question, we'd need to see the full spec. Seemingly simple features can generate a lot of complexity. Also, to answer the question in a useful way, I'd need a way to describe "how hard".

> Are other companies interested in this feature as well? Is there someone investing on it already? 

We've discussed this with Superlist in the past. @miguelcmedeiros do you know if Superlist currently has any interest in table support?

> Could my company invest for the development and integration of this feature?

Absolutely. If you haven't already, please check out www.FlutterBountyHunters.com to see our rates. We'd be happy to add your company as one of the funders and build out table support.

> Would be feasible for an external team to add this feature on the project, or may be fork it?

Anyone can fork this project, if they want to. However, I wouldn't recommend it. Our goal with Super Editor is to help the community stop fragmenting the same solution over and over. Ideally, we would build out the solution right here, as a community, sharing in Super Editor as a tool.

**Comment by shady-ti** [comment]
1. Has there been any word/progress on this?
2. What is your stand on stand-alone contributions (individuals, not companies)?

**Comment by matthew-carroll** [comment]
> Has there been any word/progress on this?

No. None of our funding companies seem to need this feature. And it's a big feature, so it's unlikely that we're going to build it without funding.

> What is your stand on stand-alone contributions (individuals, not companies)?

In general, we're happy to accept them, given the following criteria:

- The work implements something that we definitely want
- The work adheres to our coding practices, so that we don't spend a lot of time rehashing it in PRs
- The work is small enough that we can review it in a trivial amount of time

The issue with this particular feature is that it's big. It will require some sophisticated API design and a number of PRs. I don't think there's a way to effectively review that work without investing significant time on our end. Therefore, I think it's important that we find funding for this.

That said, if you can find any company or companies that would like to fund it, we're always happy to bring in new funders.

**Comment by Dboy233** [comment]
I also have a need for "Table". Because I see that there are already apps that use SuperEditor to implement the "Table" requirement. But the app is not open source.
Recently I've been working on ways to implement "Table". If possible, I hope the authors and contributors can help me and give me an implementation plan.

**Comment by rajrushilmakkar** [comment]
Hi @matthew-carroll 

We at Teachally are interested in sponsoring the development of table support in Super Editor. Could you reach out to us at danielb@teachally.com (our CEO, who will be the sponsor) and rushil@teachally.com to discuss this further?

Looking forward to your response!
